src/gui/vector/qgsattributesformmodel.cpp

```diff
@@ -0,0 +1,1162 @@
+      editorItem->setData( NodeDataRole, itemData );
+      break;
+    }
+
+    case Qgis::AttributeEditorType::Invalid:
+    {
+      QgsDebugError( QStringLiteral( "Not loading invalid attribute editor type..." ) );
+      break;
+    }
+  }
+
+  parent->addChildNode( std::move( editorItem ) );
+}
+
+QVariant QgsAttributesFormLayoutModel::data( const QModelIndex &index, int role ) const
+{
+  if ( !index.isValid() )
+    return QVariant();
+
+  const QgsAttributesFormTreeNode *node = nodeForIndex( index );
+  if ( !node )
+    return QVariant();
+
+  switch ( role )
+  {
+    case Qt::DisplayRole:
+    {
+      return node->displayName().isEmpty() ? node->name() : node->displayName();
+    }
+
+    case Qt::ToolTipRole:
+    {
+      if ( node->type() == QgsAttributesFormTreeData::Field )
+      {
+        return node->name();
+      }
+      return QVariant();
+    }
+
+    case Qt::DecorationRole:
+      return node->icon();
+
+    case Qt::BackgroundRole:
+    {
+      if ( node->type() == QgsAttributesFormTreeData::Container )
+        return QBrush( Qt::lightGray );
+
+      return QVariant();
+    }
+
+    case NodeDataRole:
+    case NodeFieldConfigRole:
+    case NodeNameRole:
+    case NodeIdRole:
+    case NodeTypeRole:
+      return node->data( role );
+
+    default:
+      return QVariant();
+  }
+}
+
+bool QgsAttributesFormLayoutModel::setData( const QModelIndex &index, const QVariant &value, int role )
+{
+  if ( !index.isValid() )
+    return false;
+
+  if ( role == NodeFieldConfigRole ) // This model doesn't store data for that role
+    return false;
+
+  QgsAttributesFormTreeNode *node = nodeForIndex( index );
+  bool result = node->setData( role, value );
+
+  if ( result )
+    emit dataChanged( index, index, { role } );
+
+  return result;
+}
+
+bool QgsAttributesFormLayoutModel::removeRows( int row, int count, const QModelIndex &parent )
+{
+  if ( row < 0 )
+    return false;
+
+  QgsAttributesFormTreeNode *node = nodeForIndex( parent );
+
+  if ( row > node->childCount() - count )
+    return false;
+
+  beginRemoveRows( parent, row, row + count - 1 );
+  for ( int r = 0; r < count; ++r )
+    node->deleteChildAtIndex( row );
+  endRemoveRows();
+  return true;
+}
+
+bool QgsAttributesFormLayoutModel::removeRow( int row, const QModelIndex &parent )
+{
+  beginRemoveRows( parent, row, row );
+  QgsAttributesFormTreeNode *node = nodeForIndex( parent );
+  node->deleteChildAtIndex( row );
+  endRemoveRows();
+  return true;
+}
+
+Qt::DropActions QgsAttributesFormLayoutModel::supportedDragActions() const
+{
+  return Qt::MoveAction;
+}
+
+Qt::DropActions QgsAttributesFormLayoutModel::supportedDropActions() const
+{
+  return Qt::DropAction::CopyAction | Qt::DropAction::MoveAction;
+}
+
+bool QgsAttributesFormLayoutModel::dropMimeData( const QMimeData *data, Qt::DropAction action, int row, int column, const QModelIndex &parent )
+{
+  Q_UNUSED( column )
+  bool bDropSuccessful = false;
```

Comment:
What does b stand for in bDrop? I'd remove the b or expand into its full meaning :)